src/Request.php

```diff
@@ -678,6 +678,30 @@ function ( $data ) {
 	 * @throws \Exception
 	 */
 	public function execute_http() {
+		// Validate content-type header for POST requests
+		if ( isset( $_SERVER['REQUEST_METHOD'] ) && 'POST' === $_SERVER['REQUEST_METHOD'] ) {
+			// Check both possible content-type header locations
+			$content_type = '';
+			if ( isset( $_SERVER['CONTENT_TYPE'] ) ) {
+				$content_type = sanitize_text_field( $_SERVER['CONTENT_TYPE'] );
+			} elseif ( isset( $_SERVER['HTTP_CONTENT_TYPE'] ) ) {
+				$content_type = sanitize_text_field( $_SERVER['HTTP_CONTENT_TYPE'] );
+			}
+
+			if ( empty( $content_type ) || 0 !== strpos( $content_type, 'application/json' ) ) {
```

Comment:
```suggestion
			if ( 'application/json' !== trim( strtolower( $content_type ) ) ) {
```

Reply:
Updated! 🙌🏻

Reply:
@justlevine had to revert this change as it causes failing test: https://github.com/wp-graphql/wp-graphql/actions/runs/13168466543/job/36753997929#step:6:584

some clients (I've seen postman send these headers, for example) send headers like: 

`Content-type: application/json; charset=utf-8` which we should not return a 415 for